bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -1173,20 +1178,95 @@ public void smartHomeCommand(String entityId, String action, @Nullable String pr
         JsonObject parameters = new JsonObject();
         parameters.addProperty("action", action);
         if (property != null) {
-            if (value instanceof QuantityType<?>) {
-                parameters.addProperty(property + ".value", ((QuantityType<?>) value).floatValue());
-                parameters.addProperty(property + ".scale",
-                        ((QuantityType<?>) value).getUnit().equals(SIUnits.CELSIUS) ? "celsius" : "fahrenheit");
-            } else if (value instanceof Boolean) {
-                parameters.addProperty(property, (boolean) value);
-            } else if (value instanceof String) {
-                parameters.addProperty(property, (String) value);
-            } else if (value instanceof Number) {
-                parameters.addProperty(property, (Number) value);
-            } else if (value instanceof Character) {
-                parameters.addProperty(property, (Character) value);
-            } else if (value instanceof JsonElement) {
-                parameters.add(property, (JsonElement) value);
+            if ("setThermostatMode".equals(action)) {
+                if (value instanceof StringType) {
+                    parameters.addProperty(property + ".value", value.toString());
+                }
+            } else if ("setTargetTemperature".equals(action)) {
+                if ("targetTemperature".equals(property)) {
+                    if (value instanceof QuantityType<?>) {
+                        parameters.addProperty(property + ".value", ((QuantityType<?>) value).floatValue());
+                        parameters.addProperty(property + ".scale",
+                                ((QuantityType<?>) value).getUnit().equals(SIUnits.CELSIUS) ? "celsius" : "fahrenheit");
+                    }
+                } else {
+                    Map<String, JsonArray> devices = null;
+                    try {
+                        devices = getSmartHomeDeviceStatesJson(new HashSet<>(getSmarthomeDeviceList()));
+                    } catch (URISyntaxException e) {
+                        logger.error("{}", e.toString());
+                    }
+                    Object applianceId = "AAA_SonarCloudService_00QAbw7vzYhdndv3-Sirp40Zargm3orQ0tKQK9sIKbqIM6-rFm3KsAnOKWApPNqtmzGL6cOR30mQrIwEQtDA";
```

Comment:
This section I am querying the current item states for the lowersetpoint and uppersetpoint because we have to feed both values into the API call regardless of which value is changing. I've made a small performance update and will commit it soon. I appreciate all of your time looking over my submission.

Reply:
Actually I meant the string itself ("AAA_SonarCloudService_00QAbw7" etc.), where does it come from?

Reply:
That is the ApplianceID for the Amazon Smart Thermostat. Should I include that in constants?

Reply:
A constant with a descriptive name could definitely help explain it in the code. So it's copied from some documentation and this is a global appliance id?

Reply:
I could not verify this, so I have scrapped it and added another method to SmartHomeBaseDevice called findEntityId() to return the entityId rather than the applianceId. This is used to quickly filter the deviceList via stream filter.

entityId is a required input parameter for smartHomeCommand(), so we use that to match the smart thermostat entityId in the full list of devices returned by getSmarthomeDeviceList().